caikit/runtime/http_server/http_server.py

```diff
@@ -607,30 +714,105 @@ async def _generator() -> pydantic_response:
                         "code": error_code,
                         "id": uuid.uuid4().hex,
                     }
-                except (CaikitCoreException, CaikitRuntimeException) as err:
-                    error_code = STATUS_CODE_TO_HTTP.get(err.status_code, 500)
-                    error_content = {
-                        "details": err.message,
-                        "code": error_code,
-                        "id": err.id,
-                    }
-                    log.error("<RUN53234506E>", error_content, exc_info=True)
-                except Exception as err:  # pylint: disable=broad-exception-caught
-                    error_code = 500
-                    error_content = {
-                        "details": f"Unhandled exception: {str(err)}",
-                        "code": error_code,
-                        "id": uuid.uuid4().hex,
-                    }
-                    log.error("<RUN51891206E>", error_content, exc_info=True)
-
-                # If an error occurs, yield an error response and terminate
+                except Exception as err:
+                    if (error_content := self._handle_exception(err)) is None:
+                        raise
                 yield ServerSentEvent(
                     data=json.dumps(error_content), event=StreamEventTypes.ERROR.value
                 )
 
             return EventSourceResponse(_generator())
 
+    #############
+    ## Helpers ##
+    #############
+
+    @staticmethod
+    def _handle_exception(err: Exception) -> Optional[dict]:
```

Comment:
This new function consolidates all error handling to a single location for consistency